sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/ControlFlowGraph/UcfgInstructionBuilder.cs

```diff
@@ -80,192 +82,242 @@ public Instruction Create(SyntaxNode syntaxNode)
             {
-                var variable = semanticModel.GetDeclaredSymbol(variableDeclarator);
-                if (IsLocalVarOrParameterOfTypeString(variable))
+                if (property.IsStatic)
+                {
+                    yield return CreateStaticCallExpression(property.ContainingType);
+                }
+                else
                 {
-                    return CreateAssignment(
-                        variableDeclarator,
-                        variable.Name,
-                        variableDeclarator.Initializer.Value);
+                    yield return ThisExpression;
                 }
             }
+        }
+
+        private IEnumerable<Instruction> CreateFromVariableDeclarator(VariableDeclaratorSyntax variableDeclarator)
+        {
+            if (variableDeclarator.Initializer == null)
+            {
+                return Enumerable.Empty<Instruction>();
+            }
 
-            return null;
+            var variable = semanticModel.GetDeclaredSymbol(variableDeclarator);
+            if (IsLocalVarOrParameter(variable))
+            {
+                return new[] { CreateAssignment(variableDeclarator, variable.Name,
+                    GetMappedExpression(variableDeclarator.Initializer.Value)) };
+            }
+
+            return Enumerable.Empty<Instruction>();
         }
 
-        private Instruction CreateFromBinaryExpression(BinaryExpressionSyntax binaryExpression) =>
-            CreateConcatenation(binaryExpression, binaryExpression.Right, binaryExpression.Left);
+        private IEnumerable<Instruction> CreateFromBinaryExpression(BinaryExpressionSyntax binaryExpression) =>
+             new[] { CreateConcatenation(binaryExpression, GetMappedExpression(binaryExpression.Right),
+                 GetMappedExpression((binaryExpression.Left))) };
 
-        private Instruction CreateFromInvocationExpression(InvocationExpressionSyntax invocationExpression)
+        private IEnumerable<Instruction> CreateFromInvocationExpression(InvocationExpressionSyntax invocationExpression)
         {
             var methodSymbol = GetSymbol(invocationExpression) as IMethodSymbol;
             if (methodSymbol == null)
             {
                 return CreateConstant(invocationExpression);
             }
 
-            var arguments = BuildArguments(invocationExpression, methodSymbol).ToArray();
+            return new[] { AddMethodCall(invocationExpression, methodSymbol, BuildArguments().ToArray()) };
+
+            IEnumerable<Expression> BuildArguments()
+            {
+                if (methodSymbol.IsStatic ||
+                    methodSymbol.ReducedFrom != null)
+                {
+                    yield return CreateStaticCallExpression(methodSymbol.ContainingType);
+                }
+
+                if (!methodSymbol.IsStatic)
+                {
+                    if (invocationExpression.Expression is MemberAccessExpressionSyntax memberAccess)
```

Comment:
I think this code should be applied to properties

Reply:
I am not sure to understand what you mean